projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -1308,32 +2478,76 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.restoreHighlight();
     }
 
+    /**
+     * ```typescript
+     * this.grid.clearGrouping(); // Clears all grouping in the grid.
+     * this.grid.clearGrouping("ID"); // Clear grouping from a particular column.
+     * ```
+     */
     public clearGrouping(name?: string): void {
         this.gridAPI.clear_groupby(this.id, name);
         this.calculateGridSizes();
 
         this.restoreHighlight();
     }
 
+    /**
+     * Returns if a group is expanded or not.
+     * ```typescript
+     * public groupRow: IGroupByRecord;
+     * const expandedGroup = this.grid.isExpandedGroup(this.groupRow);
+     * ```
+     */
     public isExpandedGroup(group: IGroupByRecord): boolean {
         const state: IGroupByExpandState = this._getStateForGroupRow(group);
         return state ? state.expanded : this.groupsExpanded;
     }
 
+    /**
+     * Toggles the expansion state of a group.
+     * ```typescript
+     * public groupRow: IGroupByRecord;
+     * const toggleExpGroup = this.grid.toggleGroup(this.groupRow);
+     * ```
+     */
     public toggleGroup(groupRow: IGroupByRecord) {
         this._toggleGroup(groupRow);
     }
 
+    /**
+     * Returns whether the record is group by.
```

Comment:
Returns if the object implements the `IGroupByRecord` interface.

Except it doesn't because the check is for internal purposes and JS doesn't support interfaces :)

I'd probably hide this